include/glow/ExecutionEngine/ExecutionEngine2.h

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_EXECUTIONENGINE_EXECUTIONENGINE2_H
+#define GLOW_EXECUTIONENGINE_EXECUTIONENGINE2_H
+
+#include "glow/Backend/Backend.h"
+#include "glow/Backend/CompiledFunction.h"
+#include "glow/Backends/DeviceManager.h"
+#include "glow/Base/Train.h"
+#include "glow/Base/Traits.h"
+#include "glow/Graph/Graph.h"
+#include "glow/Graph/PlaceholderBindings.h"
+#include "glow/Runtime/HostManager/HostManager.h"
+
+#include "llvm/ADT/ArrayRef.h"
+
+#include <memory>
+#include <unordered_map>
+
+namespace glow {
+
+/// This is the ExecutionEngine. It owns the Graph, the backend, and the
+/// compiled function.  The Graph, etc in this class are defined as pointers, in
+/// order to erase the type and prevent the internal types from leaking out to
+/// the users of this class.
+class ExecutionEngine final {
```

Comment:
Using the same name for the ExecutionEngine2 means that this will have namespace collisions with the old one, effectively theres no way to have a test suite that uses either interchangeably. Consequences:
* Must migrate every test in a file at once, can't do piece by piece.
* New tests in a file that hasn't been migrated have to use the old EE, can't use any features of the new one.

Might be less total effort to actually name this `ExecutionEngine2` and code mod it back when removing the old one. Alternatively could move the old one to a new deprecated namespace so it's clear it's old?

Reply:
Renaming to ExecutionEngine2 seems fine, I generally have been porting the whole file at a time but options are good.

Reply:
renamed